### PR TITLE
Add button platform to OpenTherm Gateway

### PR DESCRIPTION
**Pull request by mvn23**
## Proposed change

Add documentation for the newly added `button` platform to OpenTherm Gateway.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [x] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#125185
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Added support for a button interface in the OpenTherm Gateway integration, expanding device management options.
  
- **Documentation**
	- Updated the integration documentation to include a new section on the button functionality and streamlined device type information.

### CODE REVIEW

File: source/_integrations/opentherm_gw.markdown [hunk 1]
```diff
@@ -13,6 +14,7 @@ ha_codeowners:
   - '@mvn23'
 ha_domain: opentherm_gw
 ha_platforms:
+  - button
   - binary_sensor
```
Comment (c0ffeeca7) [inline source/_integrations/opentherm_gw.markdown:19]: Before the actions section, please add a section ## Entities, ### Buttons, listing the supported buttons with name and description.

Comment (mvn23) [reply source/_integrations/opentherm_gw.markdown:19]: At the moment, the only button is a `Restart` button on the device that gets restarted when the button is pressed. Do we really need documentation for that?

Comment (c0ffeeca7) [reply source/_integrations/opentherm_gw.markdown:19]: Fair question. The documentation is more helpful if all the available entities are described. The PR description mentioned that there will be more buttons to come. Also, the documentation should also be there for the other entities, such as the binary sensors, but those are outside the scope of this PR.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @mvn23 :+1:

**Pull request merged**